examples/notebooks/tour-of-beam/dataframes.ipynb

```diff
@@ -0,0 +1,744 @@
+{
+  "nbformat": 4,
+  "nbformat_minor": 2,
+  "metadata": {
+    "colab": {
+      "name": "Beam DataFrames",
+      "provenance": [],
+      "collapsed_sections": [],
+      "toc_visible": true
+    },
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    },
+    "language_info": {
+      "name": "python"
+    }
+  },
+  "cells": [
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "source": [
+        "#@title ###### Licensed to the Apache Software Foundation (ASF), Version 2.0 (the \"License\")\n",
+        "\n",
+        "# Licensed to the Apache Software Foundation (ASF) under one\n",
+        "# or more contributor license agreements. See the NOTICE file\n",
+        "# distributed with this work for additional information\n",
+        "# regarding copyright ownership. The ASF licenses this file\n",
+        "# to you under the Apache License, Version 2.0 (the\n",
+        "# \"License\"); you may not use this file except in compliance\n",
+        "# with the License. You may obtain a copy of the License at\n",
+        "#\n",
+        "#   http://www.apache.org/licenses/LICENSE-2.0\n",
+        "#\n",
+        "# Unless required by applicable law or agreed to in writing,\n",
+        "# software distributed under the License is distributed on an\n",
+        "# \"AS IS\" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY\n",
+        "# KIND, either express or implied. See the License for the\n",
+        "# specific language governing permissions and limitations\n",
+        "# under the License."
+      ],
+      "outputs": [],
+      "metadata": {
+        "cellView": "form",
+        "id": "rz2qIC9IL2rI"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# Beam DataFrames\n",
+        "\n",
+        "<button>\n",
+        "  <a href=\"https://beam.apache.org/documentation/dsls/dataframes/overview/\">\n",
+        "    <img src=\"https://beam.apache.org/images/favicon.ico\" alt=\"Open the docs\" height=\"16\"/>\n",
+        "    Beam DataFrames overview\n",
+        "  </a>\n",
+        "</button>\n",
+        "\n",
+        "Beam DataFrames provide a pandas-like [DataFrame](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html)\n",
+        "API to declare Beam pipelines.\n",
+        "\n",
+        "> ℹ️ To learn more about Beam DataFrames, take a look at the\n",
+        "[Beam DataFrames overview](https://beam.apache.org/documentation/dsls/dataframes/overview) page.\n",
+        "\n",
+        "First, we need to install Apache Beam with the `interactive` extra for the Interactive runner."
+      ],
+      "metadata": {
+        "id": "hDuXLLSZnI1D"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "source": [
+        "%pip install --quiet apache-beam[interactive]"
```

Comment:
```suggestion
        "First, we need to install Apache Beam with the `interactive` extra for the Interactive runner, as well as a supported version of pandas."
      ],
      "metadata": {
        "id": "hDuXLLSZnI1D"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "source": [
        "%pip install --quiet apache-beam[interactive],pandas==1.2.4"
```

Could we make this explicitly install pandas? Technically the original command would work since interactive beam requires pandas, but it's good to make users aware that the Beam DataFrame API requires pandas.

Reply:
How about we just mention that the Beam DataFrame API requires pandas, but let pip figure out which version to use? Pinning this to a specific version would create a maintenance burden of having to update it whenever it changes, and we currently don't have a testing infrastructure to validate this.

Reply:
My reasoning for pinning a specific version is that we need to do some work to verify the DataFrame API with new minor versions of pandas (e.g. see #15525, which will verify with pandas 1.3.x), before we can be sure the DataFrame API will work as intended with that version. You're right though that pinning it here is a bad idea.

Really we should have a separate "dataframe" extra to track this requirement, and tell users to install `apache-beam[interactive,dataframe]` here (and wherever else we talk about the requirement). I filed BEAM-12906 for this.

Reply:
Sounds good, I'll leave it like this for now, and once we have an `apache-beam[dataframe]`, we can update this notebook :)